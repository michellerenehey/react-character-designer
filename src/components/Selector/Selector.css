.selector {
  grid-column: 1/2;
  grid-row: 1/2;
}

div {
  margin: 20px;
}

.choices {
  display: flex;
  flex-direction: column;
}

button {
  width: 100px;
  background-color: var(--color3);
  color: var(--color5);
  align-self: center;
  margin: 4px;
}
