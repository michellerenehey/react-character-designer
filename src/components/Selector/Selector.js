import './Selector.css';

export default function Selector({
  head,
  setHead,
  pool,
  setPool,
  drink,
  setDrink,
  phrase,
  setPhrase,
  phraseList,
  setPhraseList,
}) {
  const handleClick = () => {
    setPhraseList((prevState) => [...prevState, phrase]);
    setPhrase('');
  };
  return (
    <>
      <div>
        <label>Pick your housewife</label>
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          <option value="kim">kim richards</option>
          <option value="kyle">kyle richards</option>
          <option value="lisa">lisa vanderpump</option>
        </select>
      </div>
      <div>
        <label>Pick her pool</label>
        <select value={pool} onChange={(e) => setPool(e.target.value)}>
          <option value="indoor">indoor pool</option>
          <option value="outdoor">outdoor pool</option>
          <option value="rooftop">rooftop pool</option>
        </select>
      </div>
      <div>
        <label>Pick her drink of choice</label>
        <select value={drink} onChange={(e) => setDrink(e.target.value)}>
          <option value="wine">white wine</option>
          <option value="tequila">tequila shot</option>
          <option value="champ">champers</option>
        </select>
      </div>
      <div>
        <label>Her catch phrase?</label>
        <input type="text" value={phrase} onChange={(e) => setPhrase(e.target.value)} />
        <button onClick={handleClick}>save the phrase</button>
      </div>
    </>
  );
}
