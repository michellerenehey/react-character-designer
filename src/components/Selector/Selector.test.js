import { render, screen } from '@testing-library/react';
import Selector from './Selector';

test('renders the Selector panel', () => {
  const container = render(<Selector head="lisa" pool="outdoor" drink="tequila" phrase="" />);
  expect(container).toMatchSnapshot();
});
