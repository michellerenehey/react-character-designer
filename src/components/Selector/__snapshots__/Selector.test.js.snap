// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Selector panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="selector"
      >
        <div
          class="choices"
        >
          <label>
            Pick your housewife:
          </label>
          <select>
            <option
              value="kim"
            >
              kim richards
            </option>
            <option
              value="kyle"
            >
              kyle richards
            </option>
            <option
              value="lisa"
            >
              lisa vanderpump
            </option>
          </select>
        </div>
        <div
          class="choices"
        >
          <label>
            Pick her pool:
          </label>
          <select>
            <option
              value="indoor"
            >
              indoor pool
            </option>
            <option
              value="outdoor"
            >
              outdoor pool
            </option>
            <option
              value="rooftop"
            >
              rooftop pool
            </option>
          </select>
        </div>
        <div
          class="choices"
        >
          <label>
            Pick her drink of choice:
          </label>
          <select>
            <option
              value="wine"
            >
              white wine
            </option>
            <option
              value="tequila"
            >
              tequila shot
            </option>
            <option
              value="champ"
            >
              champers
            </option>
          </select>
        </div>
        <div
          class="choices"
        >
          <label>
            Her catch phrase?
          </label>
          <input
            type="text"
            value=""
          />
          <button>
            save the phrase
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="selector"
    >
      <div
        class="choices"
      >
        <label>
          Pick your housewife:
        </label>
        <select>
          <option
            value="kim"
          >
            kim richards
          </option>
          <option
            value="kyle"
          >
            kyle richards
          </option>
          <option
            value="lisa"
          >
            lisa vanderpump
          </option>
        </select>
      </div>
      <div
        class="choices"
      >
        <label>
          Pick her pool:
        </label>
        <select>
          <option
            value="indoor"
          >
            indoor pool
          </option>
          <option
            value="outdoor"
          >
            outdoor pool
          </option>
          <option
            value="rooftop"
          >
            rooftop pool
          </option>
        </select>
      </div>
      <div
        class="choices"
      >
        <label>
          Pick her drink of choice:
        </label>
        <select>
          <option
            value="wine"
          >
            white wine
          </option>
          <option
            value="tequila"
          >
            tequila shot
          </option>
          <option
            value="champ"
          >
            champers
          </option>
        </select>
      </div>
      <div
        class="choices"
      >
        <label>
          Her catch phrase?
        </label>
        <input
          type="text"
          value=""
        />
        <button>
          save the phrase
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
